impl/asm/src/main/java/org/wildfly/transformer/asm/TransformerImpl.java

```diff
@@ -274,6 +274,18 @@ private void generateReflectionHandlingModelCode(String handlingClassPackage) {
                         // check if we generated reflection handling code yet, if not, generate it
                         String handlingClassName = handlingClassPackage + "/" + CLASS_FOR_NAME_PRIVATE_METHOD;
                         if (!generatedReflectionModelHandlingCode.contains(handlingClassName)) {
+                            
+                            // ensure that only one thread actually adds generated ReflectionModel handler for handlingClassPackage
+                            synchronized (generatedReflectionModelHandlingCode) {
```

Comment:
There's no need to synchronize on "generatedReflectionModelHandlingCode" 
as it is already concurrency friendly Set. This whole synchronized block can be 
eliminated and simplified to:

---

if (!generatedReflectionModelHandlingCode.add(handlingClassName) {
   // another thread will (or did) generate extra code 
  return;
}

---

Reply:
Mean to replace also "if statement" on line 276

Reply:
Good idea,I'll do that.